### PR TITLE
Fix: donation form errors

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

This updates how we handle donation form errors on v3 forms and adds styles to the notice.

## Affects

- Updates the error handing & styles on v3 forms.
- Moved the error handling for v2 forms closer to request.

#### Showcasing error from test gateway

Screenshot 2023-09-05 at 5 22 09 PM

---

Screenshot 2023-09-05 at 5 27 50 PM

## Testing Instructions

- To test gateway exceptions you will need to programmatically / hardcode throwing an exception in a gateway like the Test Gateway.  This should be tested on both v2 and v3 forms.
- To just see the error styling on v3 forms, use the test gateway and put "error" as your first name.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205405424043741
  - https://app.asana.com/0/0/1205435384192406

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@JasonTheAdams @kjohnson this is ready for review - whoever is available.

File: src/DonationForms/resources/registrars/templates/layouts/FormError.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { __ } from "@wordpress/i18n";
+
+export default function FormError({error}: {error: string}) {
+    return (
+        <div className="givewp-donation-form__errors">
+            <p className="givewp-donation-form__errors__description">
+                {__('The following error occurred when submitting the form:', 'give')}
+            </p>
+            <ul className="givewp-donation-form__errors__messages">
```
[the remark this answers is not in the document]
Comment (kjohnson) [reply src/DonationForms/resources/registrars/templates/layouts/FormError.tsx:9]: Or is this to match the design?

[the remark this answers is not in the document]
Comment (jonwaldstein) [reply src/DonationForms/resources/registrars/templates/layouts/FormError.tsx:9]: @kjohnson yea this just matches the design which displays the error in a list.

**Review by kjohnson — APPROVED** [review]
Looks good to me.

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**